tools/Iwyu/boost-all.imp

```diff
@@ -1,37 +1,86 @@
-# This file is `boost-all.imp` from the include-what-you-use repository at
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# This file is algorithmically generated to contain useful mappings for the
+# boost version installed in our container, i.e. boost version 1.71
+#
+# We generate the file following the guidlines in `boost-all.imp` from the
+# include-what-you-use repository at
 # https://github.com/include-what-you-use/include-what-you-use/tree/0.14
-# This file is originally distributed under the LLVM license.
+#
+# To recap, the steps are as follows:
+# 1. grab the output from this command,
+#    $ cd /usr/include && grep -r --exclude-dir={detail,impl} '^ *# *include' boost/ | perl -nle 'm/^([^:]+).*["<]([^>]+)[">]/ && print qq@    { include: ["<$2>", private, "<$1>", public ] },@' | grep -e \/detail\/ -e \/impl\/ | grep -e \\[\"\<boost/ | sort -u
+# 2. manually delete:
+#    $ sed '/workarounds*\.hpp/d' -i boost-all.imp
+#    Note: this step may no longer be necessary on recent boost versions?
+# 3. manually map all headers from these 5 dirs:
+#    {bind, format, filesystem, function, python}
```

Comment:
can any of this be simplified because we no longer use boost-python?

Reply:
We could omit the python mappings altogether, which would be a very minor simplification to the "code" in this file (a one line change), but add complexity to the instructions for regenerating the file. Do you want me to do this?

Reply:
Let's leave as-is then. I think being able to more easily move to newer IWYU versions is important since hopefully at some point it'll be good enough to use regularly

Reply:
Sounds good!